src/categorical_algebra/FinCats.jl

```diff
@@ -371,8 +374,11 @@ FinDomFunctor(ob_map, ::Nothing, dom::FinCat, codom::Cat) =
 function hom_map(F::FinDomFunctor, path::Path)
   C, D = dom(F), codom(F)
   path = decompose(C, path)
-  mapreduce(e -> hom_map(F, e), (gs...) -> compose(D, gs...),
-            edges(path), init=id(D, ob_map(F, src(path))))
+  #this is horrific, to be discussed
```

Comment:
Good evidence that we should incorporate "null" morphisms in some more explicit way.